pkg/operator/ceph/object/objectstore.go

```diff
@@ -99,59 +105,79 @@ func reconcileRealm(context *Context, serviceIP string, port int32) error {
 		defaultArg = "--default"
 	}
 
-	// create the realm if it doesn't exist yet
-	output, err := runAdminCommand(context, "realm", "get")
-	if err != nil {
+	// obviously need to change this
+	if isMultisite(spec) {
 		updatePeriod = true
-		output, err = runAdminCommand(context, "realm", "create", defaultArg)
+		realmID := spec.Multisite.Realm
+		realmArg = fmt.Sprintf("--rgw-realm=%s", realmID)
+		zoneGroupID := spec.Multisite.ZoneGroup
+		zoneGroupArg = fmt.Sprintf("--rgw-zonegroup=%s", zoneGroupID)
+		zoneID := spec.Multisite.Zone
+		zoneArg = fmt.Sprintf("--rgw-zone=%s", zoneID)
+		logger.Infof("Adding object store to realm %s, zonegroup=%s, zone=%s", realmID, zoneGroupID, zoneID)
+		// possibly need default arg?
+		_, err = runAdminCommandNoRealm(context, "zone", "modify", realmArg, zoneGroupArg, zoneArg, endpointArg)
 		if err != nil {
-			return errors.Wrapf(err, "failed to create rgw realm %s", context.Name)
+			// need to change this logging
+			return errors.Wrapf(err, "failed to add object store %s in rgw zone %s in zonegroup %s in realm %s", context.Name, zoneID, zoneGroupID, realmID)
+		}
+	} else {
+		// create the realm if it doesn't exist yet
+		output, err := runAdminCommand(context, "realm", "get")
+		if err != nil {
```

Comment:
Can we detect the "doesn't exist" error? Otherwise, any error would cause us to attempt to create the realm again.

Reply:
Since this is the section of the code that runs the normal object store code I'm not inclined to change it in this PR. Refactoring this function is something that should be done in a future PR.